notification/ldbstorage.go

```diff
@@ -0,0 +1,159 @@
+package notification
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net/url"
+	"strconv"
+	"sync"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+// LevelDB storage
+type LevelDBStorage struct {
+	db       *leveldb.DB
+	wg       sync.WaitGroup
+	latestID uint64
+	capacity uint64
+}
+
+func NewLevelDBStorage(dsn string, opts *opt.Options, capacity uint64) (Storage, error) {
+	url, err := url.Parse(dsn)
+	if err != nil {
+		return nil, err
+	}
+
+	// Open the database file
+	db, err := leveldb.OpenFile(url.Path, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	ldbstorage := &LevelDBStorage{db: db, capacity: capacity}
+	ldbstorage.latestID, err = ldbstorage.fetchLatestID()
+	if err != nil {
+		return nil, fmt.Errorf("error fetching the latest ID: %w", err)
+	}
+	return ldbstorage, nil
+}
+func (s *LevelDBStorage) add(event Event) error {
```

Comment:
this function does more than it should, considering its name.

Reply:
changed the function name.  Structure/interface is renamed to EventQueue to match with the functionalities